### PR TITLE
[SPARK-45027][PYTHON] Hide internal functions/variables in `pyspark.sql.functions` from auto-completion

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Hide internal functions/variables in `pyspark.sql.functions` from auto-completion

### Why are the changes needed?
to hide internal functions/variables which can be confusing, e.g. the internal help functions `to_str`, `get_active_spark_context`

before this PR:

image

image

after this PR:
image

image

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
manually check

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: python/pyspark/sql/utils.py [hunk 1]
```diff
@@ -179,7 +179,7 @@ def is_remote() -> bool:
     return "SPARK_CONNECT_MODE_ENABLED" in os.environ
 
 
-def try_remote_functions(f: FuncT) -> FuncT:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/utils.py:182]: you could just leave this as is, and fix `functions.py` when you import with an alias. e.g., `from pyspark.sql.utils import try_remote_functions as _try_remote_functions`

Comment (zhengruifeng) [reply python/pyspark/sql/utils.py:182]: ok

**Review by HyukjinKwon — APPROVED** [review]

**Review by HyukjinKwon — COMMENTED** [review]
I think you can remove `__all__` in this case.

**Comment by zhengruifeng** [comment]
> I think you can remove `__all__` in this case.

yeah, when we make sure all internal fields starts with `_`, the `__all__` is no longer needed, will update

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/functions.py [hunk 2]
```diff
@@ -55,11 +55,13 @@
 
 # Keep pandas_udf and PandasUDFType import for backwards compatible import; moved in SPARK-28264
 from pyspark.sql.pandas.functions import pandas_udf, PandasUDFType  # noqa: F401
+
+# TODO: should directly rename the internal functions
```
Comment (HyukjinKwon) [inline python/pyspark/sql/functions.py:59]: It's actually fine to don't rename. In practice, we only name underscore when the file is exposed as an API. `pyspark.sql.utils` isn't documented so it's fine. If those methods are used in other (internal) modules, we can just don't add `_` prefix.

Comment (zhengruifeng) [reply python/pyspark/sql/functions.py:59]: got it, let me remove it

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Pull request closed without merging**